katacomb/katacomb/aips_export.py

```diff
@@ -271,34 +271,37 @@ def export_calibration_solutions(uv_files, kat_adapter, mfimage_params, telstate
     for si, uv_file in enumerate(uv_files):
         try:
             with uv_factory(aips_path=uv_file, mode='r') as uvf:
-                try:
-                    log.info('Extracting phase only self-calibration from AIPS SN: %d' % (pSN))
-                    uvf.attach_table("AIPS SN", pSN)
+                log.info('Extracting self-calibration solutions from %s' % uv_file)
+                sn_versions = [table[0] for table in uvf.tablelist if table[1] == 'AIPS SN']
+                max_sn = max(sn_versions)
+                # Make sure we have a sequential integer sequence of SN table versions
+                assert sorted(sn_versions) == list(range(1, max_sn + 1))
+                pSN_get = min(max_sn, pSN)
+                if pSN_get == 0:
+                    raise ValueError('No self-calibration solutions available')
+                log.info('Extracting phase only self-calibration from AIPS SN: %d' % (pSN_get))
+                uvf.attach_table("AIPS SN", pSN_get)
+                sntab = uvf.tables["AIPS SN"]
+                # Only export dual-pol solutions
+                if sntab.keywords["NO_POL"] < 2:
+                    raise ValueError("Only dual-pol self-calibration solutions are supported")
+                for timestamp, gain in zip(*_massage_gains(sntab, ant_ordering)):
+                    ts.add('product_GPHASE', gain,
+                           ts=katdal_timestamps(timestamp, kat_adapter.midnight))
+
+                # Only try and export amp+phase solutions if we expect them
+                if apSN > 0:
```

Comment:
I like a [Happy Path](https://en.wikipedia.org/wiki/Happy_path) approach, which suggests that it is better to write:
```Python
# Skip amp+phase solutions if we don't expect them
if apSN == 0:
    continue

apSN_get = min(max_sn, pSN + apSN)
...
```
This reduces the indentation of the amp+phase block.

Reply:
Yes. OK.